### PR TITLE
Update GitHub Actions workflow

### PR DESCRIPTION
**Pull request by smorimoto**
I made some changes to keep the action more low-level.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -79,8 +82,9 @@ jobs:
         with:
           ocaml-version: ${{ matrix.ocaml-version }}
           dune-cache: true
-          lint-fmt: true
-          opam-disable-sandboxing: true
+          mode: lint-fmt
```
Comment (shonfeder) [inline .github/workflows/main.yml]: What's the effect of running `link-fmt` mode? (I see that we also have to run the format check ourselves explicitly, so just wondering what the `mode` does.)

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -99,5 +103,44 @@ jobs:
...
           dune-cache: true
-          lint-opam: true
-          opam-disable-sandboxing: true
+          mode: lint-opam
+
+      - run: opam install . --deps-only
+
+      - run: opam exec -- opam-dune-lint
+
+  deploy-doc:
```
Comment (shonfeder) [inline .github/workflows/main.yml:110]: This is a great addition. Thanks!

**Review by shonfeder — COMMENTED** [review]
Awesome! Thanks for adding in the doc deployment! That's a huge help.

I have one question about the new API, just asking as a user :)

Let me know whenever you're happy here and I'll merge it in!

**Review by shonfeder — APPROVED** [review]

**Comment by smorimoto** [comment]
I initially thought of adding mode to simplify things, but it does not seem as useful as I thought. So I decided to just remove all lint related inputs except opam-lint. I'm still wondering if even opam lint is worth keeping.

**Comment by smorimoto** [comment]
If you specify opam files with globbing pattern, running opam lint only on the root directory can miss packages in deep directories, so I keep that input just to avoid it, but I feel it's the user's responsibility.

File: .github/workflows/main.yml (hunk 1 above)
Comment (smorimoto) [reply .github/workflows/main.yml]: The mode had the functionality to find the vendored ocamlformat or to retrieve the required version from the ocamlformat config file to install ocamlformat.

**Comment by shonfeder** [comment]
Interesting dilemma! I can see the value in being able to have ecosystem standard ways of checking format/package/docs, and even publishing the docs in CI. But I'm not sure how much complexity it adds for you as the dev/maintainer, and/or how many people will need more flexibility in any case.

One question, while we're talking features, would it perhaps be more familiar if the wildcard in the ocaml version numbers was a `*` instead of a `x`? I think that would match other usage in github actions.

**Comment by smorimoto** [comment]
The maintenance difficulty is not a big problem, but I thought it probably deviated a lot from the goal of *setup*-ocaml. We are considering releasing another collection of utility action separately and I think we can include that in it. (So I'm going to remove the opam-lint input and update PR.)
I agree `*` is more familiar, but `x` is one of the most widely used pattern in the official actions. (setup-node, etc...) So I chose it.

**Comment by shonfeder** [comment]
@smorimoto releasing as a separate action makes sense to me! Thanks for explaining.

**Comment by smorimoto** [comment]
I just created the repository: https://github.com/actions-ml/actions

**Comment by smorimoto** [comment]
I will publish some utility actions here (e.g. actions-ml/actions/opam-lint, actions-ml/actions/deploy-doc, etc...)

**Comment by shonfeder** [comment]
I look forward to seeing what shows up on `actions`! Many thanks for the CI improvements :)

**Pull request merged**